### PR TITLE
Schemeful SameSite article and updated references

### PR DESCRIPTION
**Pull request by rowan-m**
Content landing for #3542 Draft fully reviewed and LGTMed.

Changes proposed in this pull request:

- New article for Schemeful SameSite
- Updated references to existing SameSite content

### CODE REVIEW

**Comment by rowan-m** [comment]
The Alex report is picking up the existing articles, which is good and I should address those, but as a separate PR.

File: src/site/_data/authorsData.json [hunk 1]
```diff
@@ -1574,5 +1574,20 @@
...
+    "name": {
+      "given": "Steven",
+      "family": "Bingler"
+    },
+    "org": {
+      "name": "Google",
+      "unit": "Chrome"
+    },
+    "descriptions": {
+      "en": "Software engineer on Chrome's improvements to anti-covert tracking efforts."
```
Comment (sbingler) [inline src/site/_data/authorsData.json:1588]: Looks like this is a bit out dated, can you change to
"Software engineer on Chrome's Privacy Sandbox."

Comment (rowan-m) [reply src/site/_data/authorsData.json:1588]: Apologies, missed that in pulling from the draft!

Comment (sbingler) [reply src/site/_data/authorsData.json]: LGTM now, thanks!

File: src/site/content/en/blog/schemeful-samesite/index.md [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+  - security
+feedback:
+  - api
+---
+
+{% Aside %}
+This article is part of a series on the `SameSite` cookie attribute changes:
+- [SameSite cookies explained](/samesite-cookies-explained/)
+- [SameSite cookies recipes](/samesite-cookie-recipes/)
+- [Schemeful SameSite](/schemeful-samesite)
```
Comment (mihajlija) [inline src/site/content/en/blog/schemeful-samesite/index.md:28]: There's a spelling discrepancy here, "Schemeful SameSite" is linking to "Schemeful Same-Site". Either is fine as long it's consistent. I suggest lower case "same-site" for the concept and using "SameSite" only for the cookie.

Comment (sbingler) [reply src/site/content/en/blog/schemeful-samesite/index.md:28]: Good catch.

"Schemeful Same-Site" is the name of this change. The concept of being the same site is "same-site" and the attribute that developers add to their cookies is "SameSite".

It's all very confusing and pedantic but that's all the more reason to stay consistent with it.

File: src/site/content/en/blog/schemeful-samesite/index.md [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+HTTPS then some of those common scenarios and the associated `SameSite` cookie
+behavior are outlined below.
+
+{% Aside 'warning' %}
+The long-term plan is to [phase out support for third-party cookies
+entirely](https://blog.chromium.org/2020/10/progress-on-privacy-sandbox-and.html),
+replacing them with privacy preserving alternatives. In scenarios where you do
+set `SameSite=None; Secure` on a cookie to allow it to be sent across schemes,
+this should only be considered a temporary solution on the migration towards
+full HTTPS.
```
Comment (mihajlija) [inline src/site/content/en/blog/schemeful-samesite/index.md:58]:
```suggestion
The long-term plan is to [phase out support for third-party cookies
entirely](https://blog.chromium.org/2020/10/progress-on-privacy-sandbox-and.html),
replacing them with privacy preserving alternatives. Setting `SameSite=None; Secure` on a cookie to allow it to be sent across schemes should only be considered a temporary solution in the migration towards
full HTTPS.
```

File: src/site/content/en/blog/schemeful-samesite/index.md [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+insecure HTTP traffic still presents an opportunity for network attackers to
+tamper with cookies that will then be used on the secure HTTPS version of the
+site. Creating this additional cross-site boundary between schemes provides
+further defense against these attacks.
+
+## Common cross-scheme scenarios
+
+{% Aside 'key-term' %}
+In the examples below where the URLs all have the same registrable domain, e.g.
+site.example, but different schemes, for example, **http**://site.example vs
```
Comment (mihajlija) [inline src/site/content/en/blog/schemeful-samesite/index.md:82]:
```suggestion
site.example, but different schemes, for example, **http**://site.example vs.
```

File: src/site/content/en/blog/schemeful-samesite/index.md [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+   <td><code>SameSite=None;Secure</code>
+   </td>
+   <td>☑️ Allowed
+   </td>
+   <td>⛔ Blocked
+   </td>
+  </tr>
+</table>
+
+### Loading Subresources
```
Comment (mihajlija) [inline src/site/content/en/blog/schemeful-samesite/index.md:139]:
```suggestion
### Loading subresources
```

File: src/site/content/en/blog/schemeful-samesite/index.md [hunk 6]
```diff
@@ -0,0 +1,377 @@
...
+{% Aside 'warning' %}
+All major browsers block [active mixed
+content](https://developer.mozilla.org/en-US/docs/Web/Security/Mixed_content)
+such as scripts or iframes. Additionally, browsers including
+[Chrome](https://blog.chromium.org/2019/10/no-more-mixed-messages-about-https.html)
+and [Firefox](https://groups.google.com/g/mozilla.dev.platform/c/F163Jz32oYY)
+are working toward upgrading or blocking passive mixed content.
+{% endAside %}
+
+Any changes you make here should only be considered a temporary fix while your
```
Comment (mihajlija) [inline src/site/content/en/blog/schemeful-samesite/index.md:150]:
```suggestion
Any changes you make here should only be considered a temporary fix while you
```

File: src/site/content/en/blog/schemeful-samesite/index.md [hunk 7]
```diff
@@ -0,0 +1,377 @@
...
+   <td>⛔ Blocked
+   </td>
+  </tr>
+</table>
+
+### POSTing a form
+
+Posting between cross-scheme versions of a website would previously allow
+cookies set with `SameSite=Lax` or `SameSite=Strict` to be sent. Now this is
+treated as a cross-site POST: only `SameSite=None` cookies can be sent. You may
```
Comment (mihajlija) [inline src/site/content/en/blog/schemeful-samesite/index.md:214]:
```suggestion
treated as a cross-site POST—only `SameSite=None` cookies can be sent. You may
```

File: src/site/content/en/blog/schemeful-samesite/index.md [hunk 8]
```diff
@@ -0,0 +1,377 @@
...
+    subresources"
+  - "Migrate entirely to HTTPS to continue allowing cookies to be set by
+    same-site subresources"
+    - A warning that the cookie **_will be_** blocked in a future version of
+      Chrome.
+  - "Migrate entirely to HTTPS to have cookies sent to same-site subresources"
+  - "Migrate entirely to HTTPS to allow cookies to be set by same-site
+    subresources"
+    - A warning that the cookie **_has been_** blocked.
+    - This warning can also appear when POSTing a form.
```
Comment (mihajlija) [inline src/site/content/en/blog/schemeful-samesite/index.md:302]:
```suggestion
- Navigation Issues
  - `Migrate entirely to HTTPS to continue having cookies sent on same-site
    requests`—A warning that the cookie **_will be_** blocked in a future version of
    Chrome.
  - `Migrate entirely to HTTPS to have cookies sent on same-site requests`—A warning 
    that the cookie **_has been_** blocked.
- Subresource loading Issues
  - `Migrate entirely to HTTPS to continue having cookies sent to same-site
    subresources` or `Migrate entirely to HTTPS to continue allowing cookies to be set by
    same-site subresources`—Warnings that the cookie **_will be_** blocked in a future 
    version of Chrome.
  - `Migrate entirely to HTTPS to have cookies sent to same-site subresources` or 
    `Migrate entirely to HTTPS to allow cookies to be set by same-site subresources`—Warnings that the cookie 
    **_has been_** blocked. The latter warning can also appear when POSTing
    a form.
```

File: src/site/content/en/blog/schemeful-samesite/index.md (hunk 2 above)
Comment (mihajlija) [reply src/site/content/en/blog/schemeful-samesite/index.md:28]: Okay, if I understand correctly, the title is referring to "same-site" as the concept but uses title case? web.dev uses sentence case for titles, so it should be "Schemeful same-site" here but we can't be pedantic on both sides :)

Comment (rowan-m) [reply src/site/content/en/blog/schemeful-samesite/index.md:28]: It is definitely a bit confusing. I've tried to ensure I'm using:

- `SameSite` when referring to the attribute
- "same-site" in prose

I would like to keep the URL as `schemeful-samesite` for consistency with the other posts. It looks like we're considering "Schemeful Same-Site" as a Proper Name though, meaning that it should be capitalised when referred to together but appears as "same-site" if we're referring just to the overall concept.

File: src/site/content/en/blog/schemeful-samesite/index.md [hunk 9]
```diff
@@ -0,0 +1,378 @@
...
+## FAQ
+
+### My site is already fully available on HTTPS, why am I seeing issues in my browser's DevTools?
+
+It's possible that some of your links and subresources still point to insecure
+URLs.
+
+One way to fix this issue is to use [HTTP
+Strict-Transport-Security](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security)
+and the includeSubDomain directive. With HSTS + `includeSubDomain` even if one
```
Comment (mihajlija) [inline src/site/content/en/blog/schemeful-samesite/index.md:325]:
```suggestion
and the `includeSubDomain` directive. With HSTS + `includeSubDomain` even if one
```

File: src/site/content/en/blog/schemeful-samesite/index.md [hunk 10]
```diff
@@ -0,0 +1,378 @@
...
+## FAQ
+
+### My site is already fully available on HTTPS, why am I seeing issues in my browser's DevTools?
+
+It's possible that some of your links and subresources still point to insecure
+URLs.
+
+One way to fix this issue is to use [HTTP
+Strict-Transport-Security](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security)
+and the `includeSubDomain` directive. With HSTS + `includeSubDomain` even if one
```
Comment (mihajlija) [inline src/site/content/en/blog/schemeful-samesite/index.md:325]:
```suggestion
One way to fix this issue is to use [HTTP
Strict-Transport-Security](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security) (HSTS)
and the `includeSubDomain` directive. With HSTS + `includeSubDomain` even if one
```

File: src/site/content/en/blog/schemeful-samesite/index.md [hunk 11]
```diff
@@ -0,0 +1,378 @@
...
+
+Cookies without a `SameSite` attribute are treated as if they specified
+`SameSite=Lax` and the same cross-scheme behavior applies to these cookies as
+well. Note that the temporary exception to unsafe methods still applies, see
+[the Lax + POST mitigation in the Chromium `SameSite`
+FAQ](https://www.chromium.org/updates/same-site/faq) for more information.
+
+### How are WebSockets affected?
+
+Websocket connections will still be considered same-site if they're the same
```
Comment (mihajlija) [inline src/site/content/en/blog/schemeful-samesite/index.md:364]:
```suggestion
WebSocket connections will still be considered same-site if they're the same
```

File: src/site/content/en/blog/schemeful-samesite/index.md [hunk 12]
```diff
@@ -0,0 +1,378 @@
...
+secureness as the page.
+
+For same-site:
+  - `wss://` ↔ `https://`
```
Comment (mihajlija) [inline src/site/content/en/blog/schemeful-samesite/index.md:368]: Accessibility tip: Could you spell out the relationship represented with this character "↔"? It's listed as "left right arrow" in Unicode which is not very helpful for anyone using a screen reader. https://www.compart.com/en/unicode/U+2194

**Review by mihajlija — APPROVED** [review]
LGTM! To save time on back and forth, I committed minor formatting edits but please do proofread to make sure you agree with them :)

File: src/site/content/en/blog/schemeful-samesite/index.md (hunk 12 above)
Comment (mihajlija) [reply src/site/content/en/blog/schemeful-samesite/index.md:368]: *this is also the reason why I changed the check mark sign in the table to one that has a simpler name

Comment (rowan-m) [reply src/site/content/en/blog/schemeful-samesite/index.md]: Replaced the character with explicit text.

File: src/site/content/en/blog/schemeful-samesite/index.md (hunk 2 above)
Comment (sbingler) [reply src/site/content/en/blog/schemeful-samesite/index.md]: I missed this until now, sorry about that.

The feature is named "Schemeful Same-Site" which is a proper noun and should have the capitalization if possible.

> I would like to keep the URL as schemeful-samesite for consistency with the other posts
What do you mean by consistent?

File: src/site/content/en/blog/schemeful-samesite/index.md [hunk 13]
```diff
@@ -0,0 +1,377 @@
...
+It's possible that some of your links and subresources still point to insecure
+URLs.
+
+One way to fix this issue is to use [HTTP
+Strict-Transport-Security](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security)
+(HSTS) and the `includeSubDomain` directive. With HSTS + `includeSubDomain` even
+if one of your pages accidentally includes an insecure link the browser will
+automatically use the secure version instead.
+
+### What if I can't upgrade to HTTPS?
```
Comment (kaycebasques) [inline src/site/content/en/blog/schemeful-samesite/index.md:328]: @rowan-m & @mihajlija I have a non-blocking requested change related to this section. See https://github.com/GoogleChrome/web.dev/issues/4278

**Review by kaycebasques — APPROVED** [review]

File: src/site/content/en/blog/schemeful-samesite/index.md (hunk 13 above)
Comment (rowan-m) [reply src/site/content/en/blog/schemeful-samesite/index.md:328]: Ack, good call. I can do some restructuring to tackle this and the Alex feedback.

**Pull request merged**